src/media/MediaAnnotator.ts

```diff
@@ -0,0 +1,132 @@
+import { Unsubscribe } from 'redux';
+import BaseAnnotator, { Options } from '../common/BaseAnnotator';
+import PopupManager from '../popup/PopupManager';
+import { DrawingManager } from '../drawing';
+import { RegionCreationManager, RegionManager } from '../region';
+import { CreatorStatus, getCreatorStatus } from '../store/creator';
+import { getAnnotation, getFileId, getIsCurrentFileVersion, getRotation } from '../store';
+import { Manager } from '../common/BaseManager';
+import './MediaAnnotator.scss';
+import { MEDIA_LOCATION_INDEX, FRAME } from '../constants';
+
+export const CSS_IS_DRAWING_CLASS = 'ba-is-drawing';
+
+export default class MediaAnnotator extends BaseAnnotator {
+    annotatedEl?: HTMLElement;
+
+    managers: Set<Manager> = new Set();
+
+    storeHandler?: Unsubscribe;
+
+    constructor(options: Options) {
+        super(options);
+
+        this.storeHandler = this.store.subscribe(this.handleStore);
+    }
+
+    destroy(): void {
+        if (this.storeHandler) {
+            this.storeHandler();
```

Comment:
why is this getting called on cleanup?

Reply:
It's used in the other annotators as well. The handler is defined in the constructor as a subscription to handle updates to the store(used to monitor when annotations are being created). Calling it again on the destroy method unsubscribes it from the store.